include/zephyr/drivers/accel2d.h

```diff
@@ -0,0 +1,680 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for GPU 2D accelerators
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_ACCEL2D_H_
+#define ZEPHYR_INCLUDE_DRIVERS_ACCEL2D_H_
+
+#include <zephyr/drivers/display.h>
+
+/**
+ * @brief ACCEL2D Driver APIs
+ * @defgroup accel2d_interface GPU 2D interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief 2D accelerator source data type
+ *
+ * Type of source data provided to 2D accelerator
+ */
+enum accel2d_src_type {
+	/** Solid color (ARGB8888 format)*/
+	ACCEL2D_SRC_COLOR,
+	/** Gradient definition used as source */
+	ACCEL2D_SRC_GRADIENT,
+};
+
+/**
+ * @brief 2D accelerator coordinate data
+ *
+ * 2D accelerator coordinate data. Given on a 2D plane, starting at upper left
+ * corner of display
+ */
+struct accel2d_coords {
+	/** X coordinate */
+	uint32_t x;
+	/** Y coordinate */
+	uint32_t y;
+};
+
+/**
+ * @brief 2D accelerator buffer descriptor
+ *
+ * Describes buffer for 2D accelerator. Note that some GPUs may have alignment
+ * restrictions on the stride or location of this buffer.
+ */
+struct accel_buf {
```

Comment:
Naming similar to `display_` wrapped descriptor?

```suggestion
struct accel2d_buffer_descriptor {
```